benchmarks/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -349,20 +351,58 @@ int main_(std::vector<std::string> const &args, const MPI_Comm comm)
     }
     }
 
-    for (int i = 0; i < n; ++i)
-      random_points_host(i) = {
-          {offset_x + random(), offset_y + random(), offset_z + random()}};
+    // Generate random points uniformly distributed within a box.
+    // FIXME: should be fixed to 1D/2D
+    auto const a = std::cbrt(n_values);
+
+    // The boxes in which the points are placed have side length two, centered
+    // around offset_[xyz] and scaled by a.
+    Kokkos::View<ArborX::Point *, DeviceType> random_points(
+        Kokkos::ViewAllocateWithoutInitializing("points"),
+        std::max(n_values, n_queries));
+    auto random_points_host = Kokkos::create_mirror_view(random_points);
+    for (int i = 0; i < random_points.extent_int(0); ++i)
+      random_points_host(i) = {{a * (offset_x + random()),
+                                a * (offset_y + random()),
+                                a * (offset_z + random())}};
     Kokkos::deep_copy(random_points, random_points_host);
+
+    Kokkos::deep_copy(
+        random_values,
+        Kokkos::subview(random_points, Kokkos::pair<int, int>(0, n_values)));
+
+    if (!shift_queries)
+    {
+      // By default, random points are "reused" between building the tree and
+      // performing queries.
+      Kokkos::deep_copy(
+          random_queries,
+          Kokkos::subview(random_points, Kokkos::pair<int, int>(0, n_queries)));
+    }
+    else
+    {
+      // For the queries, we shrink the global box by a factor three, and
+      // move it by a third of the global size towards the global center.
+      auto random_queries_host = Kokkos::create_mirror_view(random_queries);
+
+      int const max_offset = 2 * shift * i_max;
+      for (int i = 0; i < n_queries; ++i)
+        random_queries_host(i) = {
+            {a * ((offset_x + random()) / 3 + max_offset / 3),
+             a * ((offset_y + random()) / 3 + max_offset / 3),
+             a * ((offset_z + random()) / 3 + max_offset / 3)}};
+      Kokkos::deep_copy(random_queries, random_queries_host);
+    }
   }
 
   Kokkos::View<ArborX::Box *, DeviceType> bounding_boxes(
       Kokkos::ViewAllocateWithoutInitializing("bounding_boxes"), n_values);
   Kokkos::parallel_for("bvh_driver:construct_bounding_boxes",
                        Kokkos::RangePolicy<ExecutionSpace>(0, n_values),
                        KOKKOS_LAMBDA(int i) {
-                         double const x = random_points(i)[0];
-                         double const y = random_points(i)[1];
-                         double const z = random_points(i)[2];
+                         double const x = random_values(i)[0];
+                         double const y = random_values(i)[1];
+                         double const z = random_values(i)[2];
                          bounding_boxes(i) = {{{x - 1., y - 1., z - 1.}},
                                               {{x + 1., y + 1., z + 1.}}};
                        });
```

Comment:
Access traits work well for that use case.

Reply:
While true, you would replace a two lines by many more. Not really worth it.